docs/core/reminders-and-external-events.rst

```diff
@@ -0,0 +1,183 @@
+about things after a given period of time, or to respond to external events (other applications, sensors, etc.).
+You can find a full example assistant that implements these features
+`here <https://github.com/RasaHQ/rasa/tree/master/examples/reminderbot/README.md>`_.
+
+.. contents::
+   :local:
+
+
+.. _reminders:
+
+Reminders
+---------
+
+Instead of an external sensor, you might just want to be reminded about something after a certain amount of time.
+For this, Rasa provides the special event ``ReminderScheduled``, and another event, ``ReminderCancelled``, to unschedule a reminder.
+
+
+.. _scheduling-reminders-guide:
+
+Scheduling reminders
+^^^^^^^^^^^^^^^^^^^^
+
+Let's say you want your assistant to remind you to call a friend in 5 seconds.
+(You probably want some longer time span, but for the sake of testing, let it be 5 seconds.)
+Thus, we define an intent ``ask_remind_call`` with some NLU data,
+
+.. code-block:: md
+
+  ## intent:ask_remind_call
+  - remind me to call [Albert](name)
+  - remind me to call [Susan](name)
+  - later I have to call [Daksh](name)
+  - later I have to call [Anna](name)
+  ...
+
+and connect this intent with a new custom action ``action_set_reminder``.
+We could make this connection by providing training stories (recommended for more complex assistants), or using the :ref:`mapping-policy`.
+
+The custom action ``action_set_reminder`` should schedule a reminder that, 5 seconds later, triggers an intent ``EXTERNAL_reminder`` with all the entities that the user provided in his/her last message (similar to an external event):
+
+.. literalinclude:: ../../examples/reminderbot/actions.py
+   :pyobject: ActionSetReminder
+
+
+Note that this requires the ``datetime`` and ``rasa_sdk.events`` packages.
+
+Finally, we define another custom action ``action_react_to_reminder`` and link it to the ``EXTERNAL_reminder`` intent:
+
+.. code-block:: md
+
+  - EXTERNAL_reminder:
+    triggers: action_react_to_reminder
+
+where the ``action_react_to_reminder`` is
+
+.. literalinclude:: ../../examples/reminderbot/actions.py
+   :pyobject: ActionReactToReminder
+
+Instead of a custom action, we could also have used a simple response template.
+But here we want to make use of the fact that the reminder can carry entities, and we can process the entities in this custom action.
+
+.. warning::
+
+  Reminders are cancelled whenever you shutdown your Rasa server.
+
+
+.. warning::
+
+  Reminders currently (Rasa 1.8) don't work in `rasa shell`.
+  You have to test them with a
+  `running Rasa X server <https://rasa.com/docs/rasa-x/installation-and-setup/docker-compose-script/>`_ instead.
+
+
+.. note::
+
+   Proactively reaching out to the user is dependent on the abilities of a channel and
+   hence not supported by every channel. If your channel does not support it, consider
+   using the :ref:`callbackInput` channel to send messages to a webhook.
```

Comment:
Should we add a link which describes what a `webhook` is ?

Reply:
I linked the Wikipedia article.